packages/docs/src/components/Api/ApiItem.tsx

```diff
@@ -23,6 +24,11 @@ export default function ApiItem({route}: ApiItemProps): JSX.Element {
   const reflection: JSONOutput.DeclarationReflection =
     lookup[route.reflectionId];
   const [filters] = useFilters();
+  // This is needed because of some weird internal logic on
+  // Docusaurus' side of things.
+  // If you immediately just pass the filtered values
+  // the TOC will have wrong values mapped to it
+  const [passFilters, setPassFilters] = useState(false);
```

Comment:
I did a quick test and it looks like you should be able to use `isBrowser` directly, without the `useEffect` hook.
Just specify it as a dependency of `useMemo` and do:
```ts
(isBrowser && !matchFilters(filters, child))
```